pkg/tsdb/cloudwatch/get_metric_query_batches_test.go

```diff
@@ -0,0 +1,146 @@
+package cloudwatch
+
+import (
+	"testing"
+
+	"github.com/grafana/grafana/pkg/infra/log/logtest"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetMetricQueryBatches(t *testing.T) {
+	logger := &logtest.Fake{}
+	insight1 := models.CloudWatchQuery{
+		MetricQueryType: models.MetricQueryTypeQuery,
+		Id:              "i1",
+	}
+	insight2 := models.CloudWatchQuery{
+		MetricQueryType: models.MetricQueryTypeQuery,
+		Id:              "i2",
+	}
+	insight3 := models.CloudWatchQuery{
+		MetricQueryType: models.MetricQueryTypeQuery,
+		Id:              "i3",
+	}
+	metricStat := models.CloudWatchQuery{
+		MetricQueryType:  models.MetricQueryTypeSearch,
+		MetricEditorMode: models.MetricEditorModeBuilder,
+		Id:               "s1",
+	}
+	math1 := models.CloudWatchQuery{
```

Comment:
it was a little challenging to read in the tests below which queries these math queries are referencing. maybe the names could be like `m1_ref_i1`, `m3_ref_m1_m2` ? or another solution in camel case?

Reply:
done. Camel case is a more go style but I think the underscores are easier to read in this case